### PR TITLE
added a way to override where to store the mysql artifacts

### PR DESCRIPTION
**Pull request by ezraroi**
#91

### CODE REVIEW

**Comment by ezraroi** [comment]
looks like  the build failed on timeout...

File: src/main/java/com/wix/mysql/config/DownloadConfigBuilder.java [hunk 1]
```diff
@@ -4,20 +4,31 @@
...
 import de.flapdoodle.embed.process.io.progress.StandardConsoleProgressListener;
+import org.apache.commons.lang3.StringUtils;
+
+public class DownloadConfigBuilder extends
+    de.flapdoodle.embed.process.config.store.DownloadConfigBuilder {
 
-public class DownloadConfigBuilder extends de.flapdoodle.embed.process.config.store.DownloadConfigBuilder {
 
     public DownloadConfigBuilder defaults() {
+        String artifactStoreProp = System.getProperty("mysql.artifact.store");
```
Comment (noam-almog) [inline src/main/java/com/wix/mysql/config/DownloadConfigBuilder.java]: @ezraroi while this contribution may "get the job done" it's inconsistent with the current way we define the library settings. we need to properly expose this property to allow this property to be configured.
I suggest to create a new MysqlDownloadConfig that will expose the path for developers and home dir would be the default one.

**Comment by viliusl** [comment]
@ezraroi yes, I did not do any work on this library (and for some strange reason not getting notifications from github... on PRs/Issues:( ) and now I see that ci started failing:/ Looking into it in #94.

Regarding your PR - thanks, but I do agree with @noam-almog that we would like to stick to a different philosophy on configuring embedded-mysql library and make configuration changes explicit via configs.

I need to prototype a bit on this, but my idea would be:
```java
MysqldConfig mysqlDConfig = aMysqldConfig(v5_6_23)
  .withCharset(UTF8)
  .build();

StoreConfig storeConfig = aStoreConfig()
  .withCacheFolder('/tmp/.mysqld')
  .build();

NetworkConfig = networkConfig = aNetworkConfig()
  .withBaseUrl('http://my-proxy-url.com')
  .build();

EmbeddedMysql mysqld = anEmbeddedMysql(mysqlDConfig, storeConfig, networkConfig)
    .addSchema("aschema", ScriptResolver.classPathScript("db/001_init.sql"))
    .start();
```

This way:
 - configuration is explicit;
 - it is grouped by concern;
 - extensible by additional configs.

Now please give me heads-up if you want to take a stab at it and implement your concern or can wait a bit (1..2 weeks) for me to do it. Please note - tests are mandatory. This is one of the reasons why I took a detour from flapdoodle library as for one I did not like API much, but for other I wanted to roll-out features in a proper TDD fashion and be sure that what I provide as api is properly tested.

In any case - thanks for a PR!

**Comment by viliusl** [comment]
closing in favor of #96

**Pull request closed without merging**